test/integrations/testUtils.ts

```diff
@@ -104,6 +97,42 @@ export const overrideDestination = (destination: Destination, overrideConfigValu
   });
 };
 
+export const produceTestData = (testData: TestCaseData[]) => {
+  const result: any = [];
+  testData.forEach((tcData) => {
+    let events;
+    try {
+      switch (tcData.feature) {
+        case tags.FEATURES.PROCESSOR:
+          events = tcData.input.request.body;
+          break;
+        case tags.FEATURES.BATCH:
+          events = tcData.input.request.body.input;
+          break;
+        case tags.FEATURES.ROUTER:
+          events = tcData.input.request.body.input;
+          break;
+      }
+    } catch (e) {
+      throw new Error(`Error in producing test data: ${e}`);
+    }
+
+    events.map((event) => {
+      result.push(event.message);
```

Comment:
Do you think we should clean up any fields that the server may add? The payload should match an actual instrumented event.

I don't know how the server would react to those fields.

Reply:
maybe fields like `sentAt` `originalTimestamp`

Reply:
maybe fields like sentAt originalTimestamp

Reply:
In general if we don't send these fields server computes these otherwise the actual field in payload is propagated.

Reply:
Resolve this comment once you verify the server behavior with payloads containing the reserved fields.

Reply:
Added support to filter such keys